db/bigtable_eth1.go

```diff
@@ -366,57 +343,12 @@ func getBlockHandler(blocks *[]*types.Eth1BlockIndexed) func(gcp_bigtable.Row) b
 	}
 }
 
-// GetFullBlockDescending gets blocks starting at block start
-func (bigtable *Bigtable) GetFullBlockDescending(start, limit uint64) ([]*types.Eth1Block, error) {
-	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(time.Second*60))
-	defer cancel()
-
-	if start < 1 || limit < 1 || limit > start {
-		return nil, fmt.Errorf("invalid block range provided (start: %v, limit: %v)", start, limit)
-	}
-
-	startPadded := reversedPaddedBlockNumber(start)
-	endPadded := reversedPaddedBlockNumber(start - limit)
-
-	startKey := fmt.Sprintf("%s:%s", bigtable.chainId, startPadded)
-	endKey := fmt.Sprintf("%s:%s", bigtable.chainId, endPadded)
-
-	rowRange := gcp_bigtable.NewRange(startKey, endKey) //gcp_bigtable.PrefixRange("1:1000000000")
-
-	// if limit >= start { // handle retrieval of the first blocks
-	// 	rowRange = gcp_bigtable.InfiniteRange(startKey)
-	// }
-
-	rowFilter := gcp_bigtable.RowFilter(gcp_bigtable.ColumnFilter("data"))
-
-	blocks := make([]*types.Eth1Block, 0, limit)
-
-	rowHandler := func(row gcp_bigtable.Row) bool {
-		block := types.Eth1Block{}
-		err := proto.Unmarshal(row[DEFAULT_FAMILY_BLOCKS][0].Value, &block)
-		if err != nil {
-			logger.Errorf("error could not unmarschal proto object, err: %v", err)
-			return false
-		}
-		blocks = append(blocks, &block)
-		return true
-	}
-
-	// startTime := time.Now()
-	err := bigtable.tableBlocks.ReadRows(ctx, rowRange, rowHandler, rowFilter, gcp_bigtable.LimitRows(int64(limit)))
-	if err != nil {
-		return nil, err
-	}
-
-	// logger.Infof("finished getting blocks from table blocks: %v", time.Since(startTime))
-	return blocks, nil
-}
-
-/**
-* GetFullBlockDescending gets blocks ranging from high to low (both borders are inclusive)
-* high = max block number
-* low = min block number
-**/
+// GetFullBlocksDescending streams blocks ranging from high to low (both borders are inclusive) in the correct order via a channel.
+// Special handling for block 0 is implemented.
+//
+//	stream: channel the function will use for streaming
+//	high: highest (max) block number
+//	low: lowest (min) block number
```

Comment:
This causes the tooltip in VSCode to look like this:
![image](https://github.com/gobitfly/eth2-beaconchain-explorer/assets/114066135/966d109c-f3ac-4387-aa3f-085d66865263)